src/main/java/com/google/step/finscholar/firebase/FirebaseStorageManager.java

```diff
@@ -156,4 +152,130 @@ public static String getCollection(Firestore database, String collectionToGetFro
        throw new FirebaseException(ServletConstantValues.UNABLE_TO_READ_FROM_FIRESTORE, e);
     }
   }
-}
\ No newline at end of file
+
+  /**
+   * Adds support for infinite scroll by batching queries into pages of a limited size.
+   * Either we send in the first requested batch, or we send the next batch after one has already been sent.
+   * @param database - The database to retrieve from.
+   * @param collectionToGetFrom - The name of the collection to retrieve the batch from.
+   * @param batchSizeLimit - The maximum size of the batch, which is the size of a new page of data in infinite scroll.
+   * @param lastDocID - The ID of the last object I sent from the most recent batch.
+   * @param parameterToSortBy - The parameter I want to presort my collection by.
+   * @return - The json string representing the new batch of documents to be sent to the frontend.
+   * @throws FirebaseException
+   */
+  public static String getCollectionBatch(Firestore database, String collectionToGetFrom, int batchSizeLimit,  String lastDocID, String parameterToSortBy) throws FirebaseException {
+    CollectionReference collectionReference = database.collection(collectionToGetFrom);
+
+    // If the lastDocID is DEFAULT_VALUE (""), then we know this is the first batch to send.
+    // Else simply get the next batch in the collection.
+    if (lastDocID.equals(ServletConstantValues.DEFAULT_VALUE)) {
+      return getFirstBatch(collectionReference, batchSizeLimit, parameterToSortBy);
+    } else {
+      return getNextBatch(database, collectionReference, batchSizeLimit, lastDocID, parameterToSortBy);
+    }
+  }
+
+
+  /**
+   * Helper method for getCollectionBatch().
+   * Retrieve the first batch in the collection collectionReference of size batchSizeLimit.
+   * @param collectionReference - A reference to the collection I want to retrieve from.
+   * @param batchSizeLimit - The maximum size of the batch, which is the size of a new page of data in infinite scroll.
+   * @param parameterToSortBy - The parameter to sort the list of documents by.
+   * @return - The json string representing the first batch in a request for all of the documents in a collection.
+   * @throws FirebaseException
+   */
+  public static String getFirstBatch(CollectionReference collectionReference, int batchSizeLimit, String parameterToSortBy) throws FirebaseException {
+    // We have the option here to sort the collection by specific parameter beforehand (great for supporting sort-type queries later on).
+    // Setup the new Query.
+    Query page;
+    if (parameterToSortBy.equals(ServletConstantValues.DEFAULT_VALUE)) {
+      page = collectionReference.limit(batchSizeLimit);
+    } else {
+      page = collectionReference.orderBy(parameterToSortBy).limit(batchSizeLimit);
```

Comment:
You can try to use ternary operator to assign the value of page.
`page = (parameterToSortBy.equals(ServletConstantValues.DEFAULT_VALUE) ? exp1 : exp2).limit() `
not sure if the parentheses works in java, but ternary operator definitely will work.